BiweeklyMeetings/2022-05-10.md

```diff
@@ -0,0 +1,493 @@
+
+**Sebastien**-:
+So we had this idea of coming with a new derivation path with a different purpose. So if you have the root recovery phrase, then you could generate a different key path, which would be like a utility key path. With the utility key path, you could have the wallet checksum key used for the wall identifier and then a wallet encryption key use for off chain, transaction information, and so on. And so make it slightly easier for people to cope with standards in non complicated ways, instead of always relying on the master public key.
+
+**Matthias**-:
+In CIP 4, what key is used here to compute the checksum? Is it directly coming from...
+
+**Sebastien**-:
+The public key.
+
+**Matthias**-:
+The root public key as well?
+
+**Sebastien**-:
+Not the root public keys, it's the account public key.
+
+**Matthias**-:
+Okay. The account public key. Okay. So you have one checksum and per account, not per wallet in the end.
+
+**Sebastien**-:
+Yep. Because the main reason that was created was to make sure that users are restoring the right wallet. [inaudible 00:43:15] the account they were restoring is correct.
+
+**Matthias**-:
+Yeah. But CIP 30 operates at the level of wallets.
+
+**Sebastien**-:
+No, it's still account based.
+
+**Matthias**-:
+Still account based?
+
+**Sebastien**-:
+You're getting addresses for an account. You sign with an account.
+
+**Matthias**-:
+And okay. Then it's up to the wallet to decide which account I guess, because there is nothing in API that's that lets you select the account.
+
+**Sebastien**-:
+Yeah. It's when you enable CIP 30 the wallet will show you a dropdown asking you which account you want.
+
+**Matthias**-:
+Okay. So it could still work for that then.
+
+**Sebastien**-:
+Yeah.
+
+**Matthias**-:
+Okay. The idea is clear, but there is the question of making it a proper extension to CIP 30. At the same time, last time we said, we might want to sort of freeze the CIP 30 as it is now and have new extensions come as new CIPs just for the sake of having one core, which is fixed, but this thing feels also quite core in the end. So it could be just added before we freeze everything. Yeah. Okay. So from me, at least I would be quite in favor of extending the CIP 30 with a get wallet ID or get wallet checksum functions that would implement CIP 4, and have that as a the wallet ID, if it's ever needed by a DApp for any form of identification. Hmm. It's interesting 204, but we cannot actually do that with add to 5519. We can do it to convert to 5519. So we need to convert or translate the key, which is not unfeasible, but it was not quite recommended back in the days.
```

Comment:
```suggestion
Okay. The idea is clear, but there is the question of making it a proper extension to CIP 30. At the same time, last time we said, we might want to sort of freeze the CIP 30 as it is now and have new extensions come as new CIPs just for the sake of having one core, which is fixed, but this thing feels also quite core in the end. So it could be just added before we freeze everything. Yeah. Okay. So from me, at least I would be quite in favor of extending the CIP 30 with a `getWalletId` or `getWalletChecksum` functions that would implement CIP 4, and have that as a the wallet ID, if it's ever needed by a DApp for any form of identification. Hmm. It's interesting 204, but we cannot actually do that with add to 5519. We can do it to convert to 5519. So we need to convert or translate the key, which is not unfeasible, but it was not quite recommended back in the days.
```
... not recommending we must always do this, because it could add a lot of manual work to preparing the minutes... only that it's an additional thing we might check for in the transcriptions 🤔